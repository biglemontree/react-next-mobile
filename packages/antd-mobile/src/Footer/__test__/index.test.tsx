import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Footer } from '..'

const { Text, Tags, Links, Logo } = Footer

describe('Avatar', () => {
  it('footer snapshot', () => {
    const component = shallow(
      <Footer data-y="111">
        <Text>底部文案置底说明</Text>
      </Footer>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  it('footer tags snapshot', () => {
    const component = shallow(
      <Footer>
        <Text>© 2004-2020 Alipay.com. All rights reserved.</Text>
        <Tags
          tags={[
            {
              value: '蚂蚁借呗',
              onPress: () => null,
            },
            {
              value: '备用金',
              onPress: () => null,
            },
            {
              value: '花呗收钱',
              onPress: () => null,
            },
          ]}
        />
      </Footer>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  it('footer links snapshot', () => {
    const component = shallow(
      <Footer>
        <Links
          links={[
            {
              value: '蚂蚁借呗',
              onPress: () => null,
            },
            {
              value: '备用金',
              onPress: () => null,
            },
          ]}
        />
        <Text>© 2004-2020 Alipay.com. All rights reserved.</Text>
      </Footer>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  it('footer logo snapshot', () => {
    const component = shallow(
      <Footer>
        <Logo src="https://intranetproxy.alipay.com/skylark/lark/0/2020/png/35/1578040656524-338e314f-20fa-4e1e-be89-02730f286e5b.png" />
        <Text>© 2004-2020 Alipay.com. All rights reserved.</Text>
      </Footer>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })
})
