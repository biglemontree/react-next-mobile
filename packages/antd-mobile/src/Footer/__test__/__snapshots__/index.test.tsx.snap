// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar footer links snapshot 1`] = `
<div
  className="amd-footer"
>
  <Links
    links={
      Array [
        Object {
          "onPress": [Function],
          "value": "蚂蚁借呗",
        },
        Object {
          "onPress": [Function],
          "value": "备用金",
        },
      ]
    }
  />
  <Text>
    © 2004-2020 Alipay.com. All rights reserved.
  </Text>
</div>
`;

exports[`Avatar footer logo snapshot 1`] = `
<div
  className="amd-footer"
>
  <Logo
    src="https://intranetproxy.alipay.com/skylark/lark/0/2020/png/35/1578040656524-338e314f-20fa-4e1e-be89-02730f286e5b.png"
  />
  <Text>
    © 2004-2020 Alipay.com. All rights reserved.
  </Text>
</div>
`;

exports[`Avatar footer snapshot 1`] = `
<div
  className="amd-footer"
  data-y="111"
>
  <Text>
    底部文案置底说明
  </Text>
</div>
`;

exports[`Avatar footer tags snapshot 1`] = `
<div
  className="amd-footer"
>
  <Text>
    © 2004-2020 Alipay.com. All rights reserved.
  </Text>
  <Tags
    tags={
      Array [
        Object {
          "onPress": [Function],
          "value": "蚂蚁借呗",
        },
        Object {
          "onPress": [Function],
          "value": "备用金",
        },
        Object {
          "onPress": [Function],
          "value": "花呗收钱",
        },
      ]
    }
  />
</div>
`;
